### PR TITLE
Fix submesh handling in URDF importer in ROS2 Gem.

### PR DESCRIPTION
**Pull request by michalpelka**
It prevents choosing sub-mesh in a situation when the Asset processor has created more than one product asset from the source asset. Enable fbx in URDF importer.

### CODE REVIEW

**Review by cgalvan — APPROVED** [review]

**Review by AMZN-stankowi — APPROVED** [review]

File: Gems/ROS2/Code/Source/RobotImporter/Utils/SourceAssetsStorage.cpp [hunk 1]
```diff
@@ -78,7 +79,19 @@ namespace ROS2::Utils
                 t.m_assetId = info.m_assetId;
                 t.m_sourceAssetGlobalPath = fullSourcePathStr;
                 t.m_productAssetRelativePath = info.m_relativePath;
-                availableAssets[crc] = t;
+                if (availableAssets.contains(crc))
```
Comment (hultonha) [inline Gems/ROS2/Code/Source/RobotImporter/Utils/SourceAssetsStorage.cpp:82]: suggestion: This is pretty minor (feel free to ignore) but calling `contains` on the map and then using the `[]` operator is technically doing more work than is needed, we could instead use `find` and then write to the iterator if it's not equal to `end()`. Something like...

```c++
auto availableAssetIt = availableAssets.find(crc);
if (availableAssetIt != availableAssets.end()) {
    ...
    *it = availableAsset;
} else {
    availableAssets.insert({crc, availableAsset});
}
```

**Review by hultonha — APPROVED** [review]
LGTM, just have some minor feedback, nothing blocking

File: Gems/ROS2/Code/Source/RobotImporter/Utils/SourceAssetsStorage.cpp [hunk 2]
```diff
@@ -33,7 +33,8 @@ namespace ROS2::Utils
 
     AZStd::unordered_map<AZ::Crc32, AvailableAsset> GetInterestingSourceAssetsCRC()
     {
-        const AZStd::unordered_set<AZStd::string> kInterestingExtensions{ ".dae", ".stl", ".obj" };
+        const AZStd::unordered_set<AZStd::string> kInterestingExtensions{ ".dae", ".stl", ".obj", ".fbx" };
+        const AZStd::string kAzModelExtension(".azmodel");
```
[the remark this answers is not in the document]
Comment (michalpelka) [reply Gems/ROS2/Code/Source/RobotImporter/Utils/SourceAssetsStorage.cpp]: Thnaks, change to constexpr char array.

**Comment by michalpelka** [comment]
Thanks, @hultonha for reviewing. Agree with all your comment. 
Updated PR with fixes.

**Comment by hultonha** [comment]
> Thanks, @hultonha for reviewing. Agree with all your comment. Updated PR with fixes.

Awesome stuff @michalpelka, just kicked off a build

**Comment by adamdbrw** [comment]
This looks to be ready to merge and it fixes a bug, could we merge it in?

**Comment by hultonha** [comment]
Merged @adamdbrw 👍

**Pull request merged**